lib/ansible/modules/windows/win_wait_for_process.ps1

```diff
@@ -0,0 +1,152 @@
+#!powershell
+# This file is part of Ansible
+
+# Copyright (c) 2017 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+#Requires -Module Ansible.ModuleUtils.FileUtil
+
+$ErrorActionPreference = "Stop"
+
+$params = Parse-Args -arguments $args -supports_check_mode $true
+
+$process_name_exact = Get-AnsibleParam -obj $params -name "process_name_exact" -type "list" 
+$process_name_pattern = Get-AnsibleParam -obj $params -name "process_name_pattern" -type "str" 
+$process_id = Get-AnsibleParam -obj $params -name "process_id" -type "int" -default 0 #pid is a reserved variable in PowerShell.  use process_id instead.
+$owner = Get-AnsibleParam -obj $params -name "owner" -type "str"
+$sleep = Get-AnsibleParam -obj $params -name "sleep" -type "int" -default 1
+$pre_wait_delay = Get-AnsibleParam -obj $params -name "pre_wait_delay" -type "int" -default 0
+$post_wait_delay = Get-AnsibleParam -obj $params -name "post_wait_delay" -type "int" -default 0
+$process_min_count = Get-AnsibleParam -obj $params -name "process_min_count" -type "int" -default 1
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent"
+$timeout = Get-AnsibleParam -obj $params -name "timeout" -type "int" -default 300
+
+$result = @{
+    changed = $false
+}
+
+# validate the input
+if ($state -eq "absent" -and $sleep -ne 1)
+{
+    Fail-json $result "sleep parameter is of no effect when waiting for a process to stop."
```

Comment:
please consider changing 'is of no effect' to 'has no effect'. (here and below) - i think it reads more easily.  Also in situations like this its nice if the error message can tell the user how to fix things.   I.e. "Either change 'state' to 'present' or remove the 'sleep' parameter."

Reply:
I would turn this into a warning rather than a failure.